### PR TITLE
feat: added admincar command

### PR DESCRIPTION
**Pull request by t0retto**
## Description

Fills the empty "Buy Vehicle" button in the /admin menu that's already there, but does nothing.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: server/commands.lua [hunk 1]
```diff
@@ -28,6 +28,41 @@ lib.addCommand('blips', {
     TriggerClientEvent('qbx_admin:client:blips', source)
 end)
 
+lib.addCommand('admincar', {
+    help = 'Buy Vehicle',
+    restricted = config.saveVeh,
+}, function(source)
+    local src = source
```
Comment (FjamZoo) [inline server/commands.lua:35]: Don't localize source here

File: client/admin.lua [hunk 2]
```diff
@@ -499,6 +499,14 @@ RegisterNetEvent('qbx_admin:client:Show', function()
     end
 end)
 
+lib.callback.register('qbx_admin:client:GetVehicleInfo', function()
+    local vehicle = GetVehiclePedIsIn(PlayerPedId(), false)
```
Comment (FjamZoo) [inline client/admin.lua:503]: use cache.vehicle

Comment (TonybynMp4) [inline client/admin.lua:503]:
```suggestion
    local vehName = GetDisplayNameFromVehicleModel(GetEntityModel(cache.vehicle))
    local props = lib.getVehicleProperties(cache.vehicle)

    return vehName:lower(), cache.vehicle, props
```

File: server/commands.lua [hunk 3]
```diff
@@ -28,6 +28,41 @@ lib.addCommand('blips', {
...
+    local player = exports.qbx_core:GetPlayer(src)
+    local vehicles = exports.qbx_core:GetVehiclesByName()
+    local vehModel, vehicle, props = lib.callback.await('qbx_admin:client:GetVehicleInfo', src)
+
+    if vehicle == 0 then
```
Comment (FjamZoo) [inline server/commands.lua:40]: Change this to if not vehicle (cache.vehicle returns either number or false)

File: server/commands.lua
Comment (solareon) [inline server/commands.lua:1]: Could we add an option to override the ownership requirement? This could be useful when repossessing broken cars etc.

**Review by FjamZoo — CHANGES_REQUESTED** [review]

File: server/commands.lua [hunk 4]
```diff
@@ -28,6 +28,41 @@ lib.addCommand('blips', {
...
+        return exports.qbx_core:Notify(src, "Unknown vehicle, please contact your developer to register it.", 'error')
+    end
+
+    local isVehicleOwned = MySQL.scalar.await('SELECT count(*) from player_vehicles WHERE plate = ?', {props.plate})
+    
+    if isVehicleOwned > 0 then
+        return exports.qbx_core:Notify(src, "This vehicle is already owned.", 'error')
+    end
+
+    MySQL.insert(
```
Comment (TonybynMp4) [inline server/commands.lua:54]: Please use the qbx_vehicles export

File: server/commands.lua (hunk 3 above)
Comment (solareon) [reply server/commands.lua:40]:
```suggestion
    if not cache.vehicle then
```

File: client/admin.lua (hunk 2 above)
Comment (solareon) [reply client/admin.lua:503]:
```suggestion
    local vehicle =cache.vehicle
```

File: server/commands.lua (hunk 1 above)
Comment (solareon) [reply server/commands.lua:35]:
```suggestion
```

File: server/commands.lua [hunk 5]
```diff
@@ -28,6 +28,41 @@ lib.addCommand('blips', {
...
+        'INSERT INTO player_vehicles (license, citizenid, vehicle, hash, mods, plate, state) VALUES (?, ?, ?, ?, ?, ?, ?)', {
+            player.PlayerData.license,
+            player.PlayerData.citizenid, vehModel,
+            GetHashKey(vehModel),
```
[the remark this answers is not in the document]
Comment (solareon) [reply server/commands.lua:58]:
```suggestion
            joaat(vehModel),
```

**Review by solareon — CHANGES_REQUESTED** [review]
Some cleanup and refactoring

File: server/commands.lua [hunk 6]
```diff
@@ -28,6 +28,47 @@ lib.addCommand('blips', {
...
+    end
+
+    exports.qbx_vehicles:CreateVehicleEntity({
+        citizenId = player.PlayerData.citizenid,
+        model = vehModel,
+        mods = props,
+        plate = props.plate
+    })
+    ::done::
+    exports.qbx_core:Notify(source, "This vehicle is now yours.", 'success')
```
Comment (solareon) [inline server/commands.lua:69]:
```suggestion
    local vehicle = GetVehiclePedIsIn(GetPlayerPed(source))

    if not vehicle then
        return exports.qbx_core:Notify(source, "You have to be in a vehicle, to use this", 'error')
    end

    local vehModel = GetEntityModel(vehicle)

    if not exports.qbx_core:GetVehiclesByHash()[vehModel] then
        return exports.qbx_core:Notify(source, "Unknown vehicle, please contact your developer to register it.", 'error')
    end
    
    local playerData = exports.qbx_core:GetPlayer(source).PlayerData
    local vehName, props = lib.callback.await('qbx_admin:client:GetVehicleInfo', source)
    if exports.qbx_vehicles:DoesEntityPlateExist(props.plate) then
        local response = lib.callback.await('qbx_admin:client:SaveCarDialog', source)

        if not response then
            return exports.qbx_core:Notify(source, "Canceled.", 'inform')
        end
        exports.qbx_vehicles:SetVehicleEntityOwner({
            citizenId = playerData.citizenid,
            plate = props.plate
        })
    else
        exports.qbx_vehicles:CreateVehicleEntity({
            citizenId = playerData.citizenid,
            model = vehName,
            mods = props,
            plate = props.plate
        })
    end
    exports.qbx_core:Notify(source, "This vehicle is now yours.", 'success')
end)
```

This removes the source localization and cleans up the code and removing dependency on the client to not lie to us.

File: client/admin.lua [hunk 7]
```diff
@@ -499,6 +499,24 @@ RegisterNetEvent('qbx_admin:client:Show', function()
...
+        labels = {
+            confirm = 'Yes',
+            cancel = 'No'
+        }
+    })
+    return response == 'confirm' and true or false
+end)
+
+lib.callback.register('qbx_admin:client:GetVehicleInfo', function()
+    return GetDisplayNameFromVehicleModel(GetEntityModel(vehicle)):lower(), lib.getVehicleProperties(cache.vehicle)
```
Comment (solareon) [inline client/admin.lua:517]:
```suggestion
    return GetDisplayNameFromVehicleModel(GetEntityModel(cache.vehicle)):lower(), lib.getVehicleProperties(cache.vehicle)
```
My bad missed this one

**Review by solareon — APPROVED** [review]

File: client/admin.lua [hunk 8]
```diff
@@ -499,6 +499,24 @@ RegisterNetEvent('qbx_admin:client:Show', function()
...
+        header = 'Save Car',
+        content = 'This vehicle is already owned, do you want to override the current owner?',
+        centered = true,
+        cancel = true,
+        labels = {
+            confirm = 'Yes',
+            cancel = 'No'
+        }
+    })
+    return response == 'confirm' and true or false
```
Comment (TonybynMp4) [inline client/admin.lua:513]:
```suggestion
    return response == 'confirm'
```

**Review by TonybynMp4 — APPROVED** [review]
Looks good to me

File: server/commands.lua [hunk 9]
```diff
@@ -28,6 +28,44 @@ lib.addCommand('blips', {
     TriggerClientEvent('qbx_admin:client:blips', source)
 end)
 
+lib.addCommand('admincar', {
+    help = 'Buy Vehicle',
+    restricted = config.saveVeh,
+}, function(source)
+    local vehicle = GetVehiclePedIsIn(GetPlayerPed(source), false)
+    if not vehicle then
```
Comment (FjamZoo) [inline server/commands.lua:36]:
```suggestion
    if vehicle == 0 then
```

Vehicle will never be nil/false, only 0 or a number greater like 421352

File: server/commands.lua [hunk 10]
```diff
@@ -28,6 +28,44 @@ lib.addCommand('blips', {
...
+        return exports.qbx_core:Notify(source, "You have to be in a vehicle, to use this", 'error')
```
Comment (FjamZoo) [inline server/commands.lua:37]: Single quotes

File: server/commands.lua [hunk 11]
```diff
@@ -28,6 +28,44 @@ lib.addCommand('blips', {
...
+    end
+
+    local vehModel = GetEntityModel(vehicle)
+
+    if not exports.qbx_core:GetVehiclesByHash()[vehModel] then
+        return exports.qbx_core:Notify(source, "Unknown vehicle, please contact your developer to register it.", 'error')
```
Comment (FjamZoo) [inline server/commands.lua:43]: single quotes

File: server/commands.lua [hunk 12]
```diff
@@ -28,6 +28,44 @@ lib.addCommand('blips', {
...
+    end
+
+    local playerData = exports.qbx_core:GetPlayer(source).PlayerData
+    local vehName, props = lib.callback.await('qbx_admin:client:GetVehicleInfo', source)
+    if exports.qbx_vehicles:DoesEntityPlateExist(props.plate) then
+        local response = lib.callback.await('qbx_admin:client:SaveCarDialog', source)
+
+        if not response then
+            return exports.qbx_core:Notify(source, "Canceled.", 'inform')
```
Comment (FjamZoo) [inline server/commands.lua:52]: single quotes

File: server/commands.lua [hunk 13]
```diff
@@ -28,6 +28,44 @@ lib.addCommand('blips', {
...
+        })
+    else
+        exports.qbx_vehicles:CreateVehicleEntity({
+            citizenId = playerData.citizenid,
+            model = vehName,
+            mods = props,
+            plate = props.plate
+        })
+    end
+    exports.qbx_core:Notify(source, "This vehicle is now yours.", 'success')
```
Comment (FjamZoo) [inline server/commands.lua:66]: single quotes

**Review by FjamZoo — CHANGES_REQUESTED** [review]
Last few changes required, also we need you to fix the conlficts thanks.

**Review by FjamZoo — APPROVED** [review]

**Pull request merged**